x/callback/keeper/msg_server.go

```diff
@@ -21,16 +27,103 @@ func NewMsgServer(keeper Keeper) *MsgServer {
 }
 
 // CancelCallback implements types.MsgServer.
-func (*MsgServer) CancelCallback(context.Context, *types.MsgCancelCallback) (*types.MsgCancelCallbackResponse, error) {
-	panic("unimplemented 👻")
+func (s MsgServer) CancelCallback(c context.Context, request *types.MsgCancelCallback) (*types.MsgCancelCallbackResponse, error) {
+	if request == nil {
+		return nil, status.Error(codes.InvalidArgument, "empty request")
+	}
+	ctx := sdk.UnwrapSDKContext(c)
+
+	// If a callback with same job id does not exist, return error
+	callback, err := s.keeper.GetCallback(ctx, request.GetCallbackHeight(), request.GetContractAddress(), request.GetJobId())
```

Comment:
How does the authorisation look like here? Is `sender` expected to be the signer? 

If `sender` is the signer, but it supplies the `contract_address`  of a contract which is not owned what would happen?

Reply:
I see, since the authorisation check is done in the delete process, we should swap the order of execution with:

1. Delete callback
2. Do money refunds

This way we execute first the more fallible step